internal/capability/operator_install_test.go

```diff
@@ -0,0 +1,99 @@
+package capability
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/opdev/opcap/internal/operator"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Operator install tests", func() {
+	var csv operatorv1alpha1.ClusterServiceVersion
+	var fakeClient operator.Client
+	var operatorGroupData operator.OperatorGroupData
+	var subscription operator.SubscriptionData
+	var customResources []map[string]interface{}
+	var auditFn auditFn
+	var auditCleanupFn auditCleanupFn
+	BeforeEach(func() {
+		// Set up and clean up temporary directory for CSV created in operator_install
+		tmpDir, err := os.MkdirTemp("", "operator-install-*")
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(os.RemoveAll, tmpDir)
+		cwd, err := os.Getwd()
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(os.Chdir, cwd)
+		os.Chdir(tmpDir)
+
+		csv = operatorv1alpha1.ClusterServiceVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testcsv",
+				Namespace: "testns",
+			},
+			Status: operatorv1alpha1.ClusterServiceVersionStatus{
+				Phase: operatorv1alpha1.CSVPhaseSucceeded,
+			},
+		}
+
+		fakeClient = operator.NewFakeOpClient(&csv)
+
+		operatorGroupData = operator.OperatorGroupData{
+			Name:             "testog",
+			TargetNamespaces: []string{"default", "testns"},
+		}
+
+		subscription = operator.SubscriptionData{
+			Name:            "testsub",
+			Channel:         "test",
+			InstallModeType: v1alpha1.InstallModeTypeAllNamespaces,
+			Package:         "testpackage",
+			CatalogSource:   "testcatalog",
+		}
+
+		customResources = []map[string]interface{}{
+			{
+				"kind": "testkind",
+				"metadata": map[string]interface{}{
+					"name": "testname",
+				},
+			},
+		}
+
+	})
+	Context("creating a new operator install audit", func() {
+		When("given valid options", func() {
+			It("should return functional auditFn and auditCleanupFn", func() {
+				auditFn, auditCleanupFn = operatorInstall(context.TODO(),
+					withClient(fakeClient),
+					withNamespace("testingThings"),
+					withOperatorGroupData(&operatorGroupData),
+					withSubscription(&subscription),
+					withTimeout(1),
+					withCustomResources(customResources),
+				)
+				Expect(auditFn(context.TODO())).To(Succeed())
+				// test contents of fake Client actually changed/were added
+				Expect(auditCleanupFn(context.TODO())).To(Succeed())
+			})
+		})
+		When("given an invalid option to add, such as an empty namespace", func() {
```

Comment:
If this is what the test does, just shorten to "given an empty namespace"

Reply:
If you want to have the greater context, then separate it out to 

```
When("given an invalid option", func() {
    When("given an empty namespace", func() {
        ....
     }
}
```

Reply:
It would be good to eventually add more tests that exercise all the with*() Options. Up to you if you want to add more now or in a future PR.

Reply:
since those functions are in audit.go should those be in a different test for organizational clarity?

Reply:
For testing the literally with* functions, yes, that would be in audit_test.go. This should focus on what the audit itself does. And how it handles invalid values.